esiclient/tests/v1/test_node_network.py

```diff
@@ -0,0 +1,411 @@
+        self.app.client_manager.baremetal.node.set_provision_state.\
+            assert_called_once_with('node1', 'adopt')
+        self.assertEqual(
+            self.app.client_manager.baremetal.node.update.call_count, 2)
+
+    def test_take_action_adopt_no_update(self):
+        self.app.client_manager.baremetal.node.get.\
+            return_value = self.node_manageable_instance_info
+        self.app.client_manager.baremetal.port.list.\
+            return_value = [self.port1, self.port2]
+
+        def mock_node_set_provision_state(node_uuid, state):
+            self.node_manageable_instance_info.provision_state = "active"
+
+        self.app.client_manager.baremetal.node.set_provision_state.\
+            side_effect = mock_node_set_provision_state
+
+        arglist = ['node1', 'test_network']
+        verifylist = []
+
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        results = self.cmd.take_action(parsed_args)
+        expected = (
+            ["Node", "MAC Address", "Network", "Fixed IP"],
+            ["node1", "bb:bb:bb:bb:bb:bb", "test_network", "2.2.2.2"]
+        )
+        self.assertEqual(expected, results)
+        self.app.client_manager.network.create_port.\
+            assert_called_once_with(name=self.node.name,
+                                    network_id=self.network.id)
+        self.app.client_manager.baremetal.node.vif_attach.\
+            assert_called_once_with('node1', self.neutron_port.id)
+        self.app.client_manager.baremetal.node.vif_attach.\
+            assert_called_once_with('node1', self.neutron_port.id)
+        self.app.client_manager.baremetal.node.set_provision_state.\
+            assert_called_once_with('node1', 'adopt')
+        self.assertEqual(
+            self.app.client_manager.baremetal.node.update.call_count, 0)
+
+    def test_take_action_node_state_exception(self):
+        self.app.client_manager.baremetal.node.get.\
+            return_value = self.node_available
+        self.app.client_manager.baremetal.port.list.\
+            return_value = [self.port1, self.port2]
+
+        arglist = ['node1', 'test_network']
+        verifylist = []
+
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        self.assertRaisesRegex(exceptions.CommandError,
+                               'Node node1 must be in the active state',
+                               self.cmd.take_action, parsed_args)
+
+    def test_take_action_port_free_exception(self):
+        self.app.client_manager.baremetal.node.get.\
+            return_value = self.node
+        self.app.client_manager.baremetal.port.list.\
+            return_value = [self.port1]
+
+        arglist = ['node1', 'test_network']
+        verifylist = []
+
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        self.assertRaisesRegex(exceptions.CommandError,
+                               'Node node1 has no free ports',
```

Comment:
quick question because I am not too familiar with `assertRaisesRegex`: now that you have changed the output message to start with `ERROR: `, does this still work?

Reply:
Ah, it will still work (because of the the regexp), but there's no reason I can't be more specific. I'll update these checks - thanks!